sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3523,8 +3525,8 @@ class Analyzer(override val catalogManager: CatalogManager)
     val project = Project(projectList, Join(left, right, joinType, newCondition, hint))
     project.setTagValue(
       Project.hiddenOutputTag,
-      hiddenList.map(_.markAsSupportsQualifiedStar()) ++
-        project.child.metadataOutput.filter(_.supportsQualifiedStar))
+      hiddenList.map(_.markAsQualifiedAccess()) ++
```

Comment:
The new semantics read a bit weird. Previously it is understandable that `markAsSupportsQualifiedStar` means `hiddenList` can be accessed by qualified star.

But how to interpret `markAsQualifiedAccess` here? I think in `hiddenList` it is duplicated join keys. What does it mean to mark them as qualified access.

Reply:
how about `markAsQualifiedAccessOnly`?

Reply:
Hmm, sounds okay